pymeasure/instruments/hp/hp3437A.py

```diff
@@ -0,0 +1,412 @@
+
+        """
+        return self._decode_status(self, self._fetch_status(), "Number")
+
+    @number_readings.setter
+    def number_readings(self, value):
+        number_str = "N" + str(strict_range(value, [0, 9999])) + "S"
+        self.write(number_str)
+
+    @property
+    def range(self):
+        """Return the current measurement voltage range.
+
+        This property can be set, valid values: 0.1, 1, 10 (V).
+
+        .. Note::
+
+            This instrument does not have autorange capability.
+
+            Overrange will be in indicated as 0.99,9.99 or 99.9
+
+        """
+        return self._decode_range(self, self._fetch_status())
+
+    @range.setter
+    def range(self, value):
+        range_str = "R" + format(
+            round(math.log10(strict_discrete_set(value, [0.1, 1, 10])) + 2), "d"
+        )
+        self.write(range_str)
+
+    @property
+    def status(self):
+        """
+        Return an object representing the current status of the unit.
+
+        """
+        return self._decode_status(self, self._fetch_status())
+
+    @property
+    def SRQ_mask(self):
+        """Return current SRQ mask, this property can be set,
+
+        bit assigment for SRQ:
+
+        =========  ==========================
+        Bit (dec)  Description
+        =========  ==========================
+         1         SRQ when invalid program
+         2         SRQ when trigger is ignored
+         4         SRQ when data ready
+        =========  ==========================
+
+        """
+        return self._decode_status(self, self._fetch_status(), "SRQ")
+
+    @SRQ_mask.setter
+    def SRQ_mask(self, value):
+        mask_str = "E" + format(strict_range(value, [0, 7]), "o") + "S"
+        self.write(mask_str)
+
+    @property
+    def trigger(self):
+        """Return current selected trigger mode, this property can be set,
+
+        Possibe values are:
+
+        ===========  ===========================================
+        Value        Explanation
+        ===========  ===========================================
+        internal     automatic trigger (internal)
+        external     external trigger (connector on back or GET)
+        hold/manual  holds the measurement/issues a manual trigger
+        ===========  ===========================================
+
+        """
+        return self._decode_trigger(self, self._fetch_status())
+
+    @trigger.setter
+    def trigger(self, value):
+        trig_set = self.TRIGGERS[strict_discrete_set(value, self.TRIGGERS)]
+        self.write(trig_set)
+
+    # Functions using low-level access
+
+    def GPIB_trigger(self):
+        """
+        Initate trigger via low-level GPIB-command
+        (aka GET - group execute trigger)
+
+        """
+        self.adapter.connection.assert_trigger()
+
+    def reset(self):
+        """
+        Initatiates a reset (like a power-on reset) of the HP3437A
+
+        """
+        self.adapter.connection.clear()
+
```

Comment:
Does make sense to send the command "DCL" to reset the instrument ?

Reply:
`self.adapter.connection.clear()` sends `IFC`.
which works in the same way as a DCL,
sending DCL shoudl work with pyvisa and the commercial visa-backends, but pyvisa-py still has some open issues on that.

Reply:
Ok, thanks for the clarification